src/main.ts

```diff
@@ -65,12 +65,45 @@ export type ContextOptions = EventContextOptions | CallableContextOptions;
  */
 export type WrappedFunction = (data: any, options?: ContextOptions) => any | Promise<any>;
 
+/** A scheduled function that can be called with optional override values for the event context.
+ * It will subsequently invoke the cloud function it wraps with a generated event context.
+ */
+export type WrappedScheduledFunction = (options?: ContextOptions) => any | Promise<any>;
+
 /** Takes a cloud function to be tested, and returns a WrappedFunction which can be called in test code. */
-export function wrap<T>(cloudFunction: CloudFunction<T>): WrappedFunction {
+export function wrap<T>(cloudFunction: CloudFunction<T>): WrappedScheduledFunction | WrappedFunction {
   if (!has(cloudFunction, '__trigger')) {
     throw new Error('Wrap can only be called on functions written with the firebase-functions SDK.');
   }
 
+  if (get(cloudFunction, '__trigger.labels.deployment-scheduled') === 'true') {
+    const scheduledWrapped: WrappedScheduledFunction = (options: ContextOptions) => {
+      // Although in Typescript we require `options` some of our JS samples do not pass it.
+      options = options || {};
+
+      _checkOptionValidity(['eventId', 'timestamp'], options);
+      let eventContextOptions = options as EventContextOptions;
```

Comment:
lines 85 to 98 is repetitive with line 132-145. Can you factor that out? (Would involve moving the if statement in line 79 to later in this function.